app/ts/library/utilities.ts

```diff
@@ -26,25 +27,29 @@ export function removeNonStringsAndTrim(...strings: (string | boolean | undefine
 /**
  * Describe a window ethereum object
  */
-type ExternalProvider = ethers.providers.ExternalProvider
-export function assertsExternalProvider(ethereum: unknown): asserts ethereum is ExternalProvider {
-	if (ethereum === null || ethereum === undefined || typeof ethereum !== 'object') throw new Error('Ethereum object does not exist')
+type ExternalProvider = ethers.providers.ExternalProvider & { chainId: string }
+export function isExternalProvider(ethereum: unknown): ethereum is ExternalProvider {
+	return ethereum !== null && ethereum !== undefined && typeof ethereum === 'object' && 'chainId' in ethereum && typeof ethereum.chainId === 'string'
 }
 
-export function isExternalProvider(ethereum: unknown): ethereum is ExternalProvider {
-	return ethereum !== null && ethereum !== undefined && typeof ethereum === 'object'
+export function assertsExternalProvider(ethereum: unknown): asserts ethereum is ExternalProvider {
+	if (!isExternalProvider(ethereum)) throw new Error('Ethereum object does not exist')
 }
 
-export type EthereumWithListeners = {
+export type ObservableEthereum = {
 	on(eventName: string | symbol, listener: (...args: any[]) => void): void
 }
 
-export function isEthereumObservable(ethereum: unknown): ethereum is EthereumWithListeners {
+export function isEthereumObservable(ethereum: unknown): ethereum is ObservableEthereum {
 	return ethereum instanceof Object && 'on' in ethereum && typeof ethereum.on === 'function'
 }
 
+export function assertsEthereumObservable(ethereum: unknown): asserts ethereum is ObservableEthereum {
+	if (!isEthereumObservable(ethereum)) throw new Error('Ethereum object is not observable')
+}
+
 export function assertUnreachable(value: never): never {
-	throw new Error(`Never gonna give you up (${value})`)
+	throw new Error(`Unexpected code execution (${value})`)
```

Comment:
meme got old? :D

Reply:
I didn't even realize this got into the codebase until I bumped into one 😆

Reply:
So you are saying past Jubal rickrolled present Jubal?  Well done.